src/OpenTelemetry/Resources/ResourceBuilder.cs

```diff
@@ -30,30 +32,72 @@ public class ResourceBuilder
         private static readonly Resource DefaultResource;
 
         static ResourceBuilder()
+        {
+            string? serviceName = GetServiceName();
+
+            DefaultResource = new Resource(new Dictionary<string, object>
+            {
+                [ResourceSemanticConventions.AttributeServiceName] = serviceName,
+            });
+        }
+
+        private ResourceBuilder()
+        {
+        }
+
+        private static string GetServiceName()
         {
             var defaultServiceName = "unknown_service";
 
             try
             {
-                var processName = Process.GetCurrentProcess().ProcessName;
-                if (!string.IsNullOrWhiteSpace(processName))
+                var serviceName = GetGeneratedServiceName();
+                if (string.IsNullOrWhiteSpace(serviceName))
                 {
-                    defaultServiceName = $"{defaultServiceName}:{processName}";
+                    defaultServiceName = $"{defaultServiceName}:{Process.GetCurrentProcess().ProcessName}";
```

Comment:
I think this is best suited to be authored as a separate ResourceDetector, which can be either bought by the application or by auto-instrumentation.

Reply:
I think this is best suited for auto-instrumentation, it gives very little to manual instrumentation since you have easy access to the `OpenTelemetry.Resources.ResourceBuilder` anyway.

Reply:
I think this is a good plan as well. I'll close this for now and move changes into the instrumentation repo.